src/VirtualClient/VirtualClient.Actions/SysbenchOLTP/SysbenchOLTPClientExecutor.cs

```diff
@@ -0,0 +1,319 @@
+                return numberOfRuns?.ToString();
+            }
+        }
+
+        /// <summary>
+        /// The number of tables option passed to Sysbench.
+        /// </summary>
+        public string NumTables
+        {
+            get
+            {
+                return this.Parameters.GetValue<string>(nameof(SysbenchOLTPClientExecutor.NumTables));
+            }
+        }
+
+        /// <summary>
+        /// The table size option passed to Sysbench.
+        /// </summary>
+        public string RecordCount
+        {
+            get
+            {
+                return this.Parameters.GetValue<string>(nameof(SysbenchOLTPClientExecutor.RecordCount));
+            }
+        }
+
+        /// <summary>
+        /// The number of threads option passed to Sysbench.
+        /// </summary>
+        public string Threads
+        {
+            get
+            {
+                return this.Parameters.GetValue<string>(nameof(SysbenchOLTPClientExecutor.Threads));
+            }
+        }
+
+        /// <summary>
+        /// The workload option passed to Sysbench.
+        /// </summary>
+        public string Workload
+        {
+            get
+            {
+                return this.Parameters.GetValue<string>(nameof(SysbenchOLTPClientExecutor.Workload));
+            }
+        }
+
+        /// <summary>
+        /// The retry policy to apply to the client-side execution workflow.
+        /// </summary>
+        protected IAsyncPolicy ClientFlowRetryPolicy { get; set; }
+
+        /// <summary>
+        /// The timespan at which the client will poll the server for responses before
+        /// timing out.
+        /// </summary>
+        protected TimeSpan PollingTimeout { get; set; }
+
+        /// <summary>
+        /// Executes the workload.
+        /// </summary>
+        /// <param name="telemetryContext">Provides context information that will be captured with telemetry events.</param>
+        /// <param name="cancellationToken">A token that can be used to cancel the operation.</param>
+        protected override Task ExecuteAsync(EventContext telemetryContext, CancellationToken cancellationToken)
+        {
+            IPAddress ipAddress;
+            List<Task> clientWorkloadTasks = new List<Task>();
+
+            if (this.IsMultiRoleLayout())
+            {
+                IEnumerable<ClientInstance> targetServers = this.GetLayoutClientInstances(ClientRole.Server);
+
+                foreach (ClientInstance server in targetServers)
+                {
+                    clientWorkloadTasks.Add(this.ClientFlowRetryPolicy.ExecuteAsync(async () =>
+                    {
+                        if (!cancellationToken.IsCancellationRequested)
+                        {
+                            IApiClient serverApiClient = this.ApiClientManager.GetOrCreateApiClient(server.Name, server);
+
+                            // 1) Confirm server is online.
```

Comment:
Nice comments, please add the client/server workflow in the documents markdown when you got the time